src/problem3.rb

```diff
@@ -0,0 +1,33 @@
+class Problem3
+  def validateAndGetDuplicateValues(values)
+    @response = []
+    if(!values.nil? && (values.instance_of?(Array)))
+      if(values.all? {|i| i.is_a?(Integer)})
+        if(values.all? {|i| i <= values.length})
+          getDuplicateValues(values, @response)
+          return filterDistinctResponse(@response)
+        else
```

Comment:
There are 2 loops here also.

get has an `each`
filter has a `uniq` (more or less a loop depending on the implementation)

The point is that you can solve this in one loop.

Your implementation is actually close to the solution, but is very hard to read with all the extra chars, you don't need the many parentheses (if, method calls) you added and usually blocks on multiple lines are written with [do](https://stackoverflow.com/questions/2122380/using-do-block-vs-brackets#2122457)